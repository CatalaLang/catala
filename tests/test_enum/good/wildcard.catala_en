## Unit testing for wildcard in patterns

```catala
declaration enumeration E:
  -- Case1 content integer
  -- Case2
  -- Case3
```

### Simple case

```catala
declaration scope Simple_case:
  context x content E
  context y content integer

scope Simple_case:
  definition x equals Case1 content 2
  definition y equals match x with pattern
    -- Case2 : 42
    -- anything : 31
```

### Simple case 2

```catala
declaration scope Simple_case_2:
  context x content E
  context y content integer

scope Simple_case_2:
  definition x equals Case3
  definition y equals match x with pattern
    -- Case2 : 42
    -- Case1 of i : i
    -- anything : 31
```
