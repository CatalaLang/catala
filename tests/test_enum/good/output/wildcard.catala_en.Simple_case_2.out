[RESULT] Computation successful! Results:
[RESULT] x = Case3 ()
[RESULT] y = 31
